blog-posts/gatsby-ssr-caching.md

```diff
@@ -0,0 +1,98 @@
+date: "2022-02-15"
+title: "Cache Gatsby SSR pages on Gatsby Cloud"
+featuredImage: images/gatsby-ssr-caching.jpg
+author: "Fabian Dietenberger"
+authorSummary: "Senior Developer at Satellytes"
+teaserText: Learn how to cache Gatsby SSR pages on Gatsby Cloud
+seoMetaText: Learn how to cache Gatsby SSR pages on Gatsby Cloud to boost page speed. Your users won't see a difference to SSG or DSG pages anymore.
+leadboxText: Want to learn more about creating blazing fast pages? Then join us to learn more!
+attribution:
+    creator: Shiro Hatori
+    source: https://unsplash.com/photos/WR-ifjFy4CI
+---
+
+Since Gatsby 4 you can do [server side rendering](https://www.gatsbyjs.com/docs/how-to/rendering-options/using-server-side-rendering/) 
+of pages. This means, the page gets rendered at runtime, and not at buildtime.
+
+When serving a page via SSR, efficient caching becomes crucial as the browser won't show anything until the whole page 
+is rendered on the server and delivered to the browser. Caching can be done on multiple levels in your architecture, in 
+this article we will focus on caching at the edge with a CDN.
+
+If you host your Gatsby page with Gatsby Cloud, you have a CDN out of the box. The CDN is powered by Fastly, but that's 
+just an implementation detail. Caching for your static (SSG) or deferred (DSG) pages is done automatically, so no need 
+to configure anything here. But if you want to do SSR, you need to take care of the correct caching behavior by yourself.
+
+> This post was inspired by the excellent video from [Remix on Youtube](https://www.youtube.com/watch?v=bfLFHp7Sbkg) 
+> If you want to deep dive to CDN caching with SSG and SSR, you should have a look! 
+
+## The Cache Control Header
+
+Caching in Gatsby Cloud can be controlled via the HTTP caching header `Cache-Control`. This header has multiple 
+[directives](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#directives), the most importantly:
+
+- `max-age`: The time in seconds for how long the page should be cached by the browser. Use this to determine how long the page should be cached  in the browser after the user fetched it. This cache is per user.
+- `s-maxage`: The time in seconds for how long the page should be cached by the CDN. If not set, the CDN will use `max-age`. Use this to determine how long a page can be cached globally. You may want to set the time lower for new and frequently changing content and higher for old or less frequently changing content.
+- `stale-while-revalidate` : The time in seconds for how long the page should be stored by the CDN and a new page should be requested in the background when a visitor hits it. Once the new page is requests, the stale page will be replaced by it. Use this to serve an old version of a page before the new page has been rendered. This is very powerful to keep your page fast while updating it in the background. If the content is not crucial, you can set this to a very high value.
```

Comment:
The wildcard in world of CDN cache-headers! Similar to `s-maxage`-header, for the given time-period all user-requests are served instantly. But on top, the page is also refreshed in the background. If the requested page has been updated, the cached version will be replaced and all subsequent requests will serve the most up-to-date version of the content.